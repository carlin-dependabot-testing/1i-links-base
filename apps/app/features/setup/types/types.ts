import { type UseFormReturn } from 'react-hook-form'

import { type InferSettingsSchema } from '@/features/settings/schemas/settings.schema'

export interface StepProps {
  form: UseFormReturn<InferSettingsSchema>
  onNext?: () => void
  onBack?: () => void
  onSkip?: () => void
  isSkipping?: boolean
}

export interface Step {
  title: string
  description: string
}

export const SETUP_STEPS: Step[] = [
  { title: 'Welcome', description: 'Welcome to the setup wizard' },
  { title: 'Basic Settings', description: 'Configure your application basics' },
  { title: 'Request Settings', description: 'Configure link request options' },
  { title: 'Template Setup', description: 'Choose your initial data setup' }
]
