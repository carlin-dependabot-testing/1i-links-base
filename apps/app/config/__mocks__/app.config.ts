const appConfig = {
  bannerText: ''
  // Add other config properties as needed
}

export default appConfig
