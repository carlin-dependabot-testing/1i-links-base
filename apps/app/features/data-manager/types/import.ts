export type ImportResult = {
  success: {
    added: Array<{ name: string; url: string }>
    updated: Array<{ name: string; url: string }>
  }
  failed: Array<{
    name: string
    url: string
    reason: 'duplicate' | 'invalid' | 'error'
  }>
  categories: {
    new: string[]
    existing: string[]
  }
}
