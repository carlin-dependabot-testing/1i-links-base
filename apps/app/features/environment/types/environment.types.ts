import { type z } from 'zod'

import { type environmentEnum } from '@/features/settings/schemas/settings.schema'

// Derive the type from the Zod enum
export type Environment = z.infer<typeof environmentEnum>
