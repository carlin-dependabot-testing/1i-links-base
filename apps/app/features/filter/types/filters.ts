export interface FilterOption {
  id: string
  label: string
  type?: string
  icon?: React.ComponentType<{ className?: string }>
}
