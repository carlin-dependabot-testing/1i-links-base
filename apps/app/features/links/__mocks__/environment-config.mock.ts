import { type EnvironmentConfig } from '@/features/links/types/environment.types'

export const mockEnvironmentConfig: EnvironmentConfig = {
  environmentColors: {
    production: '#0f0',
    staging: '#ff0',
    integration: '#00f'
  },
  environmentLabels: {
    production: 'Production',
    staging: 'Staging',
    integration: 'Integration'
  },
  enabledEnvironments: ['production', 'staging', 'integration'],
  environmentOrder: ['production', 'staging', 'integration']
}
