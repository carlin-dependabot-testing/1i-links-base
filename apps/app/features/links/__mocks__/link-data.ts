import { type LinksApp } from '@/features/links/types/link.types'

export const mockLinkData: Record<string, LinksApp> = {
  basic: {
    name: 'Test App',
    description: 'Test Description',
    category: 'Category 1',
    color: '#000000',
    environments: {
      production: 'https://test.com'
    },
    tags: ['test-tag'],
    createdAt: '2024-03-20',
    timestamp: '2024-03-20'
  },
  withMultipleEnvironments: {
    name: 'Multi Env App',
    description: 'App with multiple environments',
    category: 'Category 2',
    color: '#000000',
    environments: {
      production: 'https://prod.test.com',
      staging: 'https://staging.test.com',
      integration: 'https://dev.test.com'
    },
    tags: ['test-tag'],
    createdAt: '2024-03-20',
    timestamp: '2024-03-20'
  }
} as const
