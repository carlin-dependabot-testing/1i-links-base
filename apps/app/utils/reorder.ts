import { type LinksApp } from '@/features/links/types/link.types'

/**
 * Reorders elements in a list by moving the element at startIndex to endIndex.
 *
 * @param {any[]} list - The list of elements.
 * @param {number} startIndex - The index of the element to move.
 * @param {number} endIndex - The index where the element should be moved to.
 * @returns {any[]} A new list with the element reordered.
 *
 * @example
 * const list = ['a', 'b', 'c', 'd', 'e'];
 * const result = reorder(list, 2, 4);
 * console.log(result); // ['a', 'b', 'd', 'e', 'c']
 */
export const reorder = (
  list: LinksApp[],
  startIndex: number,
  endIndex: number
) => {
  const result = Array.from(list)

  const [removed] = result.splice(startIndex, 1)
  result.splice(endIndex, 0, removed)

  return result
}
