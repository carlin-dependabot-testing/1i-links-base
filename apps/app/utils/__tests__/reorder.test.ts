import { mockLinkData } from '@/features/links/__mocks__/link-data'
import { type LinksApp } from '@/features/links/types/link.types'

import { reorder } from '../reorder'

describe('reorder', () => {
  // Create an array from the mock data
  const mockLinks: LinksApp[] = Object.values(mockLinkData)

  it('reorder should render correctly', () => {
    const result = reorder(mockLinks, 0, 1)
    expect(result).toMatchSnapshot()
  })

  it('should move an item from first to second position', () => {
    const result = reorder(mockLinks, 0, 1)
    expect(result[1]).toEqual(mockLinks[0])
    expect(result).toHaveLength(mockLinks.length)
  })

  it('should move an item from second to first position', () => {
    const result = reorder(mockLinks, 1, 0)
    expect(result[0]).toEqual(mockLinks[1])
    expect(result).toHaveLength(mockLinks.length)
  })

  it('should not modify the original array', () => {
    const originalArray = [...mockLinks]
    reorder(mockLinks, 0, 1)
    expect(mockLinks).toEqual(originalArray)
  })

  it('should handle same start and end index', () => {
    const result = reorder(mockLinks, 1, 1)
    expect(result).toEqual(mockLinks)
  })
})
