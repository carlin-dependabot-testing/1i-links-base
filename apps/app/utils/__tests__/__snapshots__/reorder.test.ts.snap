// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reorder reorder should render correctly 1`] = `
[
  {
    "category": "Category 2",
    "color": "#000000",
    "createdAt": "2024-03-20",
    "description": "App with multiple environments",
    "environments": {
      "integration": "https://dev.test.com",
      "production": "https://prod.test.com",
      "staging": "https://staging.test.com",
    },
    "name": "Multi Env App",
    "tags": [
      "test-tag",
    ],
    "timestamp": "2024-03-20",
  },
  {
    "category": "Category 1",
    "color": "#000000",
    "createdAt": "2024-03-20",
    "description": "Test Description",
    "environments": {
      "production": "https://test.com",
    },
    "name": "Test App",
    "tags": [
      "test-tag",
    ],
    "timestamp": "2024-03-20",
  },
]
`;
